version_differences.md

```diff
@@ -0,0 +1,395 @@
+# Version and Language Differences
+
+_Note on OS compatibility_: There is considerable variation in file name casing (e.g. `Bauhaus.bsh` vs. `BAUHAUS.BSH`) across game versions, languages, and folders. Projects targeting support for OS with case-sensitive file systems (i.e. Linux) should keep this in mind. It does not matter on Windows, since Windows' file system is case-insensitive.
+
+- [Original](#original)
+- [Neue Inseln, neue Abenteuer (NINA)](#neue-inseln-neue-abenteuer-nina)
+- [Königs-Edition (KE)](#königs-edition-ke)
+    - [COD files](#cod-files)
+    - [GFX](#gfx)
+    - [MGFX](#mgfx)
+    - [SGFX](#sgfx)
+    - [ToolGfx](#toolgfx)
+        - [Language-agnostic Files](#language-agnostic-files)
+        - [Language-specific Files](#language-specific-files)
+- [Good Old Games (GOG)](#good-old-games-gog)
+    - [English, German, French Versions](#english-german-french-versions)
+        - [COD files](#cod-files-1)
+        - [GFX/MGFX/SGFX](#gfxmgfxsgfx)
+        - [ToolGfx](#toolgfx-1)
+            - [Language-agnostic files](#language-agnostic-files-1)
+            - [Language-specific files](#language-specific-files-1)
+    - [Polish Version](#polish-version)
+        - [COD files](#cod-files-2)
+        - [GFX](#gfx-1)
+        - [MGFX](#mgfx-1)
+        - [SGFX](#sgfx-1)
+        - [ToolGfx](#toolgfx-2)
+- [History Edition (HE)](#history-edition-he)
+    - [COD files](#cod-files-3)
+    - [GFX, MGFX, SGFX](#gfx-mgfx-sgfx)
+    - [ToolGfx](#toolgfx-3)
+        - [Language-agnostic files](#language-agnostic-files-2)
+        - [Language-specific files](#language-specific-files-2)
+
+## Original
+TODO
+
+## Neue Inseln, neue Abenteuer (NINA)
+TODO
+
+## Königs-Edition (KE)
+Information taken from the German release:
+* EXE file version: `0.2.5.2`
+* EXE product version: `1.0.0.5`
+
+### COD files
+| Name | SHA-256 |
+|------|---------|
+| haeuser.cod | `82799ca9b24ae05745818f39c302f46c732e90eb7d6675d5cd6661fed15d2656` |
+| figuren.cod | `2313eeb802ae7f2c4130c611e96489bc172a451d85710d4bf842d58b9e81115c` |
+
+### GFX
+See [files/gfx/](files/gfx/) for a detailed discussion of file contents.
+
+Changes in the KE version compared to **TODO the Polish GOG version, which is yet to be identified (suspicion is NINA)**:
+* Additions: `FISCHE.BSH`, `GAUKLER.BSH`, `SCHATTEN.BSH`.
+* Modifications:
+    * `STADTFLD.BSH` adds **TODO what exactly?**
+    * `TIERE.BSH` adds an antelope
+    * in `MAEHER.BSH`, a single pixel in the 1060th image has changed
+    * in `EFFEKTE.BSH`, a volcano projectiles and crates floating on the surface (shown after a ship sinks) were added
+    * `SHIP.BSH` has a different file hash, but image contents are identical
```

Comment:
The explanation is that the KE version dropped duplicate textures from BSH files.
This is also done excessively in STADTFLD.BSH by the way, to the point where I noticed that it's faster to read unique bitmaps once, instead of naively iterating over offsets in the BSH index and reading the same textures multiple times. Faster in python, at least.